lib/ansible/modules/cloud/amazon/emr_cluster_find.py

```diff
@@ -0,0 +1,155 @@
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: emr_cluster_find
+short_description:  Find AWS Elastic Map Reduce
+version_added: "2.6"
+description:
+  - return a list of Cluster EMR based on name and state
+  - results can be sorted
+  - It depends on boto3
+options:
+  region:
+    description:
+      - region where this module will find EMR
+  state:
+    description:
+      - looking for EMR according with this Status,
+        can be one of these (starting|bootstrapping|running|waiting|terminating|terminated)
+    required: true
+  name_regex:
+    description:
+      - regexp used by matched the EMR Cluster Name
+        if this option is blank, all cluster will be matched
+  limit:
+    description:
+      - limit the number of occours is return
+  sort_order:
+    description:
+      - define results order is ascending or descending, default is ascending
+author:
+  - Wellington Moreira Ramos (@wmaramos)
+extends_documentation_fragment:
+  - aws
+  - ec2
+'''
+
+EXAMPLES = '''
+- name: Looking for EMR based on status and name
+  emr_cluster_find:
+    name_regex: '^emr-test-name'
+    region: us-east-1
+    state: running
+
+- name: Looking for EMR based on name
+  emr_cluster_find:
+    name_regex: '`emr-test-name`' # in this case regexp will be *.emr-test-name.*
+    region: us-east-1
+
+- name: Return all EMR with status RUNNING
+  emr_cluster_find:
+    status: running
+    region: us-east-1
+'''
+
+RETURN = '''
+# For more information see http://boto3.readthedocs.io/en/latest/reference/services/emr.html#EMR.Client.list_clusters
+---
+results:
+  description: dictionary containg all cluster settings see link above for more information
+  returned: success
+  type: complex
+  contains:
+    id:
+      description: The Cluster ID from EMR
+      type: str
+      sample: j-12CK7ZJ37B5YA
+    name:
+      description: name of EMR Cluster
+      type: str
+      sample: emr-test-name
+    status:
+      description: details about the current status of the cluster
+      type: dict
+'''
+
+try:
+    from botocore.exception import ClientError
+except ImportError:
+    pass  # will be picked up from imported HAS_BOTO3
+
+import re
+import traceback
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import ec2_argument_spec, boto3_conn, get_aws_connection_info
+from ansible.module_utils.ec2 import HAS_BOTO3, camel_dict_to_snake_dict
+
+
+def find_emr_clusters(client, module):
+    name_regex = module.params.get('name_regex')
+    state = module.params.get('state').upper()
+    limit = module.params.get('limit')
+    sort_order = module.params.get('sort_order')
+
+    resource = client.list_clusters(ClusterStates=[state])
```

Comment:
This should have exception handling around it. Once you update to AnsibleAWSModule you can do:
```
except (BotoCoreError, ClientError) as e:
    module.fail_json_aws(e, msg="Unable to list clusters")
```
and it will use e.response and add the traceback for you.